data_safe_haven/logging/non_logging_singleton.py

```diff
@@ -0,0 +1,14 @@
+import logging
+
+from data_safe_haven.singleton import Singleton
+
+
+class NonLoggingSingleton(logging.Logger, metaclass=Singleton):
+    """
+    Non-logging singleton that can be used by anything needing logs to be consumed
+    """
+
+    def __init__(self) -> None:
+        super().__init__(name="non-logger", level=logging.CRITICAL + 10)
+        while self.handlers:
+            self.removeHandler(self.handlers[0])
```

Comment:
Could we add some tests for this.

Things like,

- Check the logging level of the object
- Check that it is a singleton?
- Ensuring that `logger.info()` _etc._ doesn't produce and file or terminal output.

Reply:
Done in d90adf6